dask/dataframe/io/parquet/core.py

```diff
@@ -473,6 +479,22 @@ def to_parquet(
     # Trim any protocol information from the path before forwarding
     path = fs._strip_protocol(path)
 
+    if overwrite:
+        working_dir = fs.expand_path(".")[0]
+        if path in ["", ".", "./", "/", working_dir]:
+            raise ValueError(
+                "If using overwrite=True, path must be an explicit and not the current directory!!"
+            )
+        if fs.exists(path) and fs.isdir(path):
+            # If the path exists, is a directory, and there are files present to overwrite
+            pfiles = fs.ls(path)
+            if len(pfiles) > 0:
+                fs.rm(path, recursive=True)
+            else:
+                raise FileNotFoundError(
+                    "Can not overwrite empty path or path not found!!"
+                )
```

Comment:
```suggestion
        if path == working_dir:
            raise ValueError(
                "Cannot clear the contents of the current working directory!"
            )
        if append:
            raise ValueError("Cannot use both `overwrite=True` and `append=True`!")
        if fs.exists(path) and fs.isdir(path):
            # Only remove path contents if
            # (1) The path exists
            # (2) The path is a directory
            # (3) The path is not the current working directory
            fs.rm(path, recursive=True)
```

Just an example of how to implement the functionality in the suggested docstring change.